### PR TITLE
Add gem ransack + edit payment page

### PR DESCRIPTION
**Pull request by nhat15**
## Related Tickets
- [#53787](https://edu-redmine.sun-asterisk.vn/issues/53787)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2022-09-26 07-55-20
Screenshot from 2022-09-26 07-55-24

### CODE REVIEW

**Comment by nhat15** [comment]
ready

File: app/controllers/admin/admin_controller.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ class Admin::AdminController < ApplicationController
   def is_admin?
     return admin_root_path if current_user.admin?
 
-    redirect_to login_path
+    flash[:danger] = "Cannot access to this page"
```
Comment (huongnt-2545) [inline app/controllers/admin/admin_controller.rb]: I18n

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -19,3 +19,5 @@ import "@fortawesome/fontawesome-free/css/all";
 window.jQuery = $;
 window.$ = $;
 
+
```
Comment (huongnt-2545) [inline app/javascript/packs/application.js]: tự dưng thêm 2 dòng trống trong file này chi zậy e
xóa đi nhé

File: app/javascript/packs/date_picker.js [hunk 3]
```diff
@@ -24,13 +24,19 @@ $(document).on('turbolinks:load', function () {
   showNavbar('header-toggle', 'nav-bar', 'body-pd', 'header')
 
   /*===== LINK ACTIVE =====*/
-  const linkColor = document.querySelectorAll('.nav_link')
+  // const linkColor = document.querySelectorAll('.nav_link')
```
Comment (huongnt-2545) [inline app/javascript/packs/date_picker.js]: xóa code nếu ko dùng đến e nhé

File: app/javascript/packs/date_picker.js [hunk 4]
```diff
@@ -24,13 +24,19 @@ $(document).on('turbolinks:load', function () {
...
 
-  function colorLink() {
-    if (linkColor) {
-      linkColor.forEach(l => l.classList.remove('active'))
-      this.classList.add('active')
-    }
-  }
-  linkColor.forEach(l => l.addEventListener('click', colorLink))
+  // function colorLink() {
```
Comment (huongnt-2545) [inline app/javascript/packs/date_picker.js]: tương tự

File: app/views/admin/payments/_search.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<%= search_form_for [:admin, @search], class: "input-group", method: "get" do |f| %>
+  <div class = "col-12 col-sm-12 col-md-3 col-lg-3 col-xl-3 col-xxl-3">
+    <%= f.label :movie_title_cont, t(".search_by_title") %>
+    <%= f.text_field :movie_title_cont, 
+        placeholder: "Search by title", class: "form-control title-input" %>
+  </div>
+  <div class="col-12 col-sm-12 col-md-3 col-lg-3 col-xl-3 col-xxl-3">
+    <label class = "blank">.</label>
+    <br/>
+    <%= f.submit "Search", class: "btn btn-primary" %>
```
Comment (huongnt-2545) [inline app/views/admin/payments/_search.html.erb]: I18n

File: db/seeds.rb [hunk 6]
```diff
@@ -1,92 +1,92 @@
-Genre.create!([{name: 'Horror'}, {name: 'Comedy'}, {name: 'Action'}, {name: 'Adventure'}, {name: 'Fantasy'}])
+# Genre.create!([{name: 'Horror'}, {name: 'Comedy'}, {name: 'Action'}, {name: 'Adventure'}, {name: 'Fantasy'}])
```
Comment (huongnt-2545) [inline db/seeds.rb]: chạy xong rồi thì xóa comment đi em nhé

**Comment by nhat15** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**